shiny/express/_run.py

```diff
@@ -176,6 +183,9 @@ def reset_top_level_recall_context_manager() -> None:
 
 
 def get_top_level_recall_context_manager() -> RecallContextManager[Tag]:
+    if _top_level_recall_context_manager is None:
+        raise RuntimeError("No top-level recall context manager has been set.")
```

Comment:
Is there a reasonable way you expect users could hit this condition? (And if so, would this error message help them fix it?)